LibTrixi.jl/examples/libelixir_t8code3d_euler_baroclinic_instability.jl

```diff
@@ -193,38 +230,51 @@ end
 # The function to create the simulation state needs to be named `init_simstate`
 function init_simstate()
 
-    ###############################################################################
-    # Setup for the baroclinic instability test
+    # compressible euler equations
     gamma = 1.4
     equations = CompressibleEulerEquations3D(gamma)
 
-    ###############################################################################
-    # semidiscretization of the problem
-
+    # setup of the problem
     initial_condition = initial_condition_baroclinic_instability
 
     boundary_conditions = Dict(:inside => boundary_condition_slip_wall,
                                :outside => boundary_condition_slip_wall)
 
-    # This is a good estimate for the speed of sound in this example.
-    # Other values between 300 and 400 should work as well.
+    # estimate for the speed of sound
     surface_flux = FluxLMARS(340)
     volume_flux = flux_kennedy_gruber
     solver = DGSEM(polydeg = 5, surface_flux = surface_flux,
                    volume_integral = VolumeIntegralFluxDifferencing(volume_flux))
 
-    trees_per_cube_face = (16, 8)
-    mesh = Trixi.P4estMeshCubedSphere(trees_per_cube_face..., 6.371229e6, 30000.0,
-                                      polydeg = 5, initial_refinement_level = 0)
+    # for nice results, use 4 and 8 here
+    lat_lon_levels = 2
+    layers = 4
+    mesh = Trixi.T8codeMeshCubedSphere(lat_lon_levels, layers, 6.371229e6, 30000.0,
+                                       polydeg = 5, initial_refinement_level = 0)
+
+    # create the data registry and three vectors for the source terms
+    registry = LibTrixiDataRegistry(undef, 4)
+
+    nnodesdim = Trixi.nnodes(solver)
+    nnodes = nnodesdim^3
+    nelements = Trixi.ncells(mesh)
+
+    # provide some data because calc_sources! will already be called during initialization
+    zero_data = zeros(Float64, nelements*nnodes)
+    registry[1] = zero_data
+    registry[2] = zero_data
+    registry[3] = zero_data
+    registry[4] = zero_data
```

Comment:
The current implementation does not seem correct. All entries of `registry` are assigned the same vector `zero_data`, i.e., they point to a single data structure.

I see that they will be overwritten later again. However, my feeling is that it would be better to make sure that this is not accidentally considered a valid implementation:
```suggestion
    # provide some data because calc_sources! will already be called during initialization
    # Note: the data pointers in the registry will be overwritten before the first real use
    zero_data = zeros(Float64, nelements*nnodes)
    registry[1] = zero_data
    registry[2] = zero_data
    registry[3] = zero_data
    registry[4] = zero_data
```

Reply:
As a side note, I am starting to wonder if storage for source terms should maybe live on the Trixi.jl side instead of the C side. This goes against previous statements I made, and I wouldn't change it now before we have gathered more experience with it. However, it seems to me that it is overly convoluted this way, especially since it's effectively Trixi.jl that needs and controls the memory, not the C side

Reply:
True! I now allocate separate arrays to avoid confusion.

Reply:
The argument for memory management on C side was that some large software frameworks have their own storage pool and would like to use it.